### DISCUSSION TITLE
SH1122 display compatibility

### DISCUSSION THREAD

**Discussion by Renber78** [sle118/squeezelite-esp32#304] [Q&A] [answered]
Good morning,
I found a display that meets my expectations (small size) for building my SqueezeESP32.
Before ordering, can someone tell me if it has a chance of working on existing drivers?
THANKS

[https://www.alibaba.com/product-detail/Resolution-256x64-SPI-interface-white-7P_62427607341.html](url)

**Comment by sle118** [comment]
Unlikely to work out of the box, but probably not a lot of work to derive a driver from one of our supported options for someone motivated

**Comment by philippe44** [comment]
I missed that - It's an interesting display. I've ordered it and I'll do the driver. It's likely a copycat of the SSD1322 as the SH1106 is a copycat (almost) of the SSD1306

**Reply by Renber78** [reply]
I ordered one too

**Comment by philippe44** [comment, marked as answer]
Driver has been added

**Reply by Renber78** [reply]
Great !
Mine arrives next week.
Thanks for the work.

**Comment by Renber78** [comment]
Hi,
I received my screen and was able to test it.
It's okay but far from perfect...

The maximum SPI speed is 12Mhz, after which it stalls (all my other screens support 40Mhz without problem).

There are some pretty unpleasant brightness issues:
  - When displaying album art, you can clearly see stray lines in the image
- The brightness set to level 5 does not illuminate any brighter than level 4 and the stray lines are amplified.

I think (but I haven't analyzed yet) that it's a hardware problem. It seems that the internal step-up power supply (12V Oled) is not dimensioned correctly and that it fails when there are too many pixels lit and the brightness is set to 4 or 5.
I will analyze this hypothesis as quickly as possible because my screen will be hidden under a blue tinted plexiglass and I need all the lighting power.

Have you also noticed these problems with your screen?

**Reply by philippe44** [reply]
No I don't have these issues. The line might also be a refresh rate issue, I can increase it

**Reply by philippe44** [reply]
And how is it powered? Where do you take Vcc from? I've also tried up to 26 MHz and it works fine (can't go above as my I/O are not in the special SPI mux)

**Reply by Renber78** [reply]
The analysis was quickly done.
There is an LDO on the screen to do 3V3. So the screen is designed for 5V! (Thanks to the Chinese seller for not providing me with this data).
After having bridged the LDO (My power supply is 3V3) the screen works perfectly well.
I tested it at 26Mhz too and it's ok now.

So in the end, a small, easy-to-read and space-saving OLED screen, I agree!
Thanks for your help :-)
​
